### PR TITLE
Hide process runner behind cloud sdk cli

### PR DESCRIPTION
**Pull request by loosebazooka**
Expose everything a user needs directly though the CloudSdkCli object.

### CODE REVIEW

**Comment by loosebazooka** [comment]
@meltsufin @etanshaul what do you guys think. We really don't want the user to have to deal with any of our process runner abstractions. I think I need to move a few more things out of internal pkg

File: src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkCli.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+     */
+    public Builder gcloudOutputFormat(String format) {
+      this.gcloudOutputFormat = format;
+      return this;
+    }
+
+    /**
+     * An environment variable for gcloud invocations.
+     */
+    public Builder environmentVariable(Environment key, String value) {
```
Comment (etanshaul) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkCli.java]: I wonder if there's any value in hiding the environment variables from the client too. I.e. pass them in like any other gcloud parameter so the client doesn't need to know if its a flag or env var.

**Comment by meltsufin** [comment]
I think hiding the process runner stuff behind the CloudSdk will indeed make the library usage simpler.  What's the benefit of splitting CloudSdk into CloudSdk that just takes that sdk path, and the CloudSdkCli?

File: src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkCli.java [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+     */
+    public Builder async() {
+      this.async = true;
+      return this;
+    }
+
+    /**
+     * Client listener on the process output stream.
+     */
+    public Builder stdOutListener(ProcessOutputLineListener listener) {
```
Comment (etanshaul) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkCli.java]: +1 on hiding the process running and exposing the listeners like this.

**Comment by meltsufin** [comment]
@loosebazooka  Do you want to close this one, since it's taken care of by #67?

**Pull request closed without merging**